src/main/java/org/jenkinsci/plugins/configfiles/buildwrapper/ManagedFile.java

```diff
@@ -110,16 +110,24 @@ public String getDisplayName() {
             return "";
         }
 
-        public ListBoxModel doFillFileIdItems(@AncestorInPath ItemGroup context, @AncestorInPath Item project, @AncestorInPath AccessControlled ac) {
+        @POST
+        public ListBoxModel doFillFileIdItems(@AncestorInPath ItemGroup context, @AncestorInPath Item project, @AncestorInPath AccessControlled ac,
+                                              @QueryParameter String fileId) {
             // You should have permission to configure your project in order to get the available managed files
+            ListBoxModel items = new ListBoxModel();
+            items.add("please select", "");
             if (project != null) {
-                project.checkPermission(Item.CONFIGURE);
+                if (!project.hasPermission(Item.EXTENDED_READ)) {
+                    items.add(fileId);
+                    return items;
+                }
             } else {
-                ac.checkPermission(Item.CREATE); // This to get the first parent to validate the authorization (Folder, View or Jenkins)
+                if (!ac.hasPermission(Item.CONFIGURE)) { // This to get the first parent to validate the authorization (Folder, View or Jenkins)
```

Comment:
Because of my last comment. 

```suggestion
                if (!ac.hasPermission(Item.CREATE)) { // This to get the first parent to validate the authorization (Folder, View or Jenkins)
```